src/components/shared/helpers/normalizeProps.js

```diff
@@ -0,0 +1,35 @@
+const PRIMITIVE_TYPES = new Set(['string', 'number', 'boolean'])
```

Comment:
Why use a set here over Array.includes for the check below?

Reply:
I tend to use the Set datatype for looking things up like this... It is slow to initialise but fast to do the actual lookup. Not that performance would matter much, given the fact that we only have three items here, so the difference between direct access and having to iterate is not really all that relevant. It's more about finding the correct data type for what I am trying to do:
> Sets are intended to represent a unique collection of values

And that's what I am trying to represent here.

But I'd be happy to switch to an array. I think I have some more Sets in the current codebase....